src/Core/src/Graphics/PaintExtensions.iOS.cs

```diff
@@ -165,5 +165,30 @@ static NSNumber[] GetCAGradientLayerLocations(List<PaintGradientStop> gradientSt
 				return locations;
 			}
 		}
+
+		static CGColor[] GetCAGradientLayerColors(List<PaintGradientStop> gradientStops)
+		{
+			if (gradientStops == null || gradientStops.Count == 0)
+				return new CGColor[0];
+
+			CGColor[] colors = new CGColor[gradientStops.Count];
+
+			int index = 0;
+			foreach (var gradientStop in gradientStops)
+			{
+				if (gradientStop.Color == Colors.Transparent)
+				{
+					var color = gradientStops[index == 0 ? index + 1 : index - 1].Color;
```

Comment:
What happens if there are multiple transparent together? trans at 0, trans at 0.3 and then red at 1? Also, what happens if the transparent is between two colors? red, trans, green.

nit: this feels like a for loop instead of foreach. I personally don't like the rando index variable outside the loop. But I am also just weird.

Reply:
Added a new sample and a new Device Test to validate that case:
![image](https://github.com/dotnet/maui/assets/6755973/1c3483c4-e74b-4e26-814b-850cffd12051)